Mathlib/MeasureTheory/Measure/Haar/DistribChar.lean

```diff
@@ -0,0 +1,86 @@
+/-
+Copyright (c) 2024 Andrew Yang, Yaël Dillies, Javier López-Contreras. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Andrew Yang, Yaël Dillies, Javier López-Contreras
+-/
+import Mathlib.MeasureTheory.Measure.Haar.Unique
+
+/-!
+# The distributive character of Haar measures
+
+Given a group `G` acting on an measurable additive commutative group `A`, and an element `g : G`,
+one can pull back the Haar measure `μ` of `A` along the map `(g • ·) : A → A` to get another Haar
+measure `μ'` on `A`. By unicity of Haar measures, there exists some nonnegative real number `r` such
+that `μ' = r • μ`. We can thus define a map `distribHaarChar : G → ℝ≥0` sending `g` to its
+associated real number `r`. Furthermore, this number doesn't depend on the Haar measure `μ` we
+started with, and `distribHaarChar` is a group homomorphism.
```

Comment:
Why `distribHaarChar`, and not simply `HaarChar`?

Reply:
because we're crucially using that `a • (b + c) = a • b + a • c` instead of `a • (b * c) = (a • b) * c`, which would give us the existing `modularCharacter`